### PR TITLE
feat: rules of hooks

### PR DESCRIPTION
**Pull request by thymikee**
### Summary

- Adds hooks plugin
- Cleans up a bit

### Test plan

Fixup tests

### CODE REVIEW

File: test/package.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
   "main": "index.js",
   "license": "MIT",
   "dependencies": {
-    "@callstack/eslint-config": "..",
-    "eslint": "^5.4.0",
-    "react": "^16.4.2"
+    "@callstack/eslint-config": "*",
```
Comment (turnrye) [inline test/package.json]: 🤘🏻movin too fast

File: yarn.lock [hunk 2]
```diff
@@ -372,8 +372,8 @@ error-ex@^1.2.0:
 
 es-abstract@^1.11.0:
   version "1.13.0"
-  resolved "https://artifactory.autozone.com:443/artifactory/api/npm/npm/es-abstract/-/es-abstract-1.13.0.tgz#ac86145fdd5099d8dd49558ccba2eaf9b88e24e9"
```
Comment (turnrye) [inline yarn.lock]: Oh snap! Thanks! Didn't realize this made it out.

File: test/package.json (hunk 1 above)
Comment (thymikee) [reply test/package.json]: I'm linking it properly with yarn now. Using `file://` protocol results in Yarn installing projects _very_ slow (they're duplicated in cache as well). Not sure if that's desired behavior or a bug tho. Linking works in this case and it's fast enough :D

**Comment by turnrye** [comment]
❤️

File: yarn.lock (hunk 2 above)
Comment (thymikee) [reply yarn.lock]: It did because Travis had some issues with actually boot up on this project 😄

**Pull request merged**